src/rdyadvance.c

```diff
@@ -169,34 +165,143 @@ static PetscErrorCode CreateVizViewerContext(RDy rdy, ViewerContext *viz) {
       // CGNS output is handled via the Options database. We need to set monitoring
       // for all formats that aren't XDMF or CGNS.
       if (rdy->config.output.format != OUTPUT_CGNS) {
-        PetscCall(TSMonitorSet(rdy->ts, (PetscErrorCode(*)(TS, PetscInt, PetscReal, Vec, void *))TSMonitorSolution, viz->vf, NULL));
+        PetscCall(TSMonitorSet(rdy->ts, (PetscErrorCode(*)(TS, PetscInt, PetscReal, Vec, void *))TSMonitorSolution, rdy->output_vf, NULL));
       }
     }
   }
 
   PetscFunctionReturn(0);
 }
 
-PetscErrorCode RDyRun(RDy rdy) {
+// this is called when -preload is set so we can get more accurate timings
+// for the solver we're using
+static PetscErrorCode CalibrateSolverTimers(RDy rdy) {
+  PetscFunctionBegin;
+
+  // create a "preload" solution so we can advance one step
+  Vec X_preload;
+  PetscCall(VecDuplicate(rdy->X, &X_preload));
+  PetscCall(VecCopy(rdy->X, X_preload));
+
+  // set tolerances to make the calibration step cheaper
+  SNES      snes;
+  PetscReal r_tol;
+  PetscCall(TSGetSNES(rdy->ts, &snes));
+  PetscCall(SNESGetTolerances(snes, NULL, &r_tol, NULL, NULL, NULL));
```

Comment:
Presently, we aren't using SNES because we are using explicit methods. So, we should get rid of this code.

Reply:
Good point!